soh/soh/Enhancements/crowd-control/CrowdControl.cpp

```diff
@@ -260,328 +270,189 @@ CrowdControl::Effect* CrowdControl::ParseMessage(char payload[512]) {
     effect->lastExecutionResult = EffectResult::Initiate;
     effect->id = dataReceived["id"];
     auto parameters = dataReceived["parameters"];
+    auto effectName = dataReceived["code"].get<std::string>();
+
     if (parameters.size() > 0) {
-        effect->value = dataReceived["parameters"][0];
+        effect->value[0] = dataReceived["parameters"][0];
     }
-    effect->type = dataReceived["code"].get<std::string>();
 
-    if (effect->type == EFFECT_HIGH_GRAVITY || effect->type == EFFECT_LOW_GRAVITY) {
+    // Assign GameInteractionEffect + values to CC effect.
+    // Categories are mostly used for checking for conflicting timed effects.
+    if (effectName == EFFECT_ADD_HEART_CONTAINER) {
+        effect->giEffect = new GameInteractionEffect::AddHeartContainer();
+    } else if (effectName == EFFECT_REMOVE_HEART_CONTAINER) {
+        effect->giEffect = new GameInteractionEffect::RemoveHeartContainer();
+    } else if (effectName == EFFECT_FILL_MAGIC) {
+        effect->giEffect = new GameInteractionEffect::FillMagic();
+    } else if (effectName == EFFECT_EMPTY_MAGIC) {
+        effect->giEffect = new GameInteractionEffect::EmptyMagic();
+    } else if (effectName == EFFECT_ADD_RUPEES) {
+        effect->giEffect = new GameInteractionEffect::GiveRupees();
+    } else if (effectName == EFFECT_REMOVE_RUPEES) {
+        effect->giEffect = new GameInteractionEffect::TakeRupees();
+    } else if (effectName == EFFECT_NO_UI) {
+        effect->category = "ui";
```

Comment:
it might be nice to `#define` these category strings like the `effectName` strings